view/frontend/web/js/utils/when-defined.js

```diff
@@ -29,45 +29,47 @@ define([], function () {
      * @param {Object} object to check for property definition
      * @param {number|string} property that is expected to be defined on {@see object}
      * @param {Function} callback function to be called when {@see property} gets defined on {@see object}
-     * @param {null} key deprecated parameter used for setting multiple callbacks per property
+     * @param {string} key used to prevent setting duplicate callbacks 
      */
-    function whenDefined(object, property, callback, key = null) {
-        if (object.hasOwnProperty(property)) {
+    function whenDefined (object, property, callback, key) {
+        if (object.hasOwnProperty(property) && typeof object[property] !== 'undefined') {
             callback();
         } else {
             var overloadedPropertyName = '_' + property;
             if (!whenDefinedCallbacks.has(object)) {
                 whenDefinedCallbacks.set(object, new Map([]));
             }
             if (!whenDefinedCallbacks.get(object).has(property)) {
```

Comment:
Could we do an early return here? It looks like it may add readability
if (whenDefinedCallbacks.get(object).has(property)) {
 whenDefinedCallbacks.get(object).get(property).set(key, callback);
 return;
}
whenDefinedCallbacks.get(object).set(property, []).set(key, callback);

Reply:
@vitaliyreznikov done